app/scripts/hglib.js

```diff
@@ -31,21 +31,42 @@ const launch = (element, config, options) => {
    * @param {Object} options Options that affect how the component is drawn and
    * and behaves.
    *
+   * @param {Object} componentCallback A callback to be called once the component
+   * is rendered. If it's not passed in, the block until the component is created.
+   *
+   * @param {Object} promise Return the ref as a promise or not.
    * @return  {Object} The instance's public API
    */
-  let component;
+  if (refAsPromise) {
+    const p = new Promise((resolve, reject) => {
+      ReactDOM.render(
+        <HiGlassComponent
+          ref={c => resolve(c)}
+          options={options || {}}
+          viewConfig={config}
+        />,
+        element
+      );
+    });
+
+    return p;
```

Comment:
Why assigning the promise first? Can't it just be returned immediately?

Reply:
You're right. It can. Changed.